### PR TITLE
fix: improve sata controller error handling

### PR DESCRIPTION
**Pull request by tenthirtyam**
- Utilize `errors.Is` for accurate error comparison when checking for `ErrNoSataController`, allowing for correct handling of wrapped errors.
- Implement error wrapping with `%w` in `fmt.Errorf` to preserve underlying error information.
- Add handling for unexpected error conditions from `FindSATAController` to ensure robustness.

### CODE REVIEW

File: builder/vsphere/common/step_add_cdrom.go [hunk 1]
```diff
@@ -81,10 +82,15 @@ func (s *StepAddCDRom) Run(_ context.Context, state multistep.StateBag) multiste
 	vm := state.Get("vm").(driver.VirtualMachine)
 
 	if s.Config.CdromType == "sata" {
-		if _, err := vm.FindSATAController(); err == driver.ErrNoSataController {
-			ui.Say("Adding SATA controller...")
-			if err := vm.AddSATAController(); err != nil {
-				state.Put("error", fmt.Errorf("error adding SATA controller: %v", err))
+		if _, err := vm.FindSATAController(); err != nil {
+			if errors.Is(err, driver.ErrNoSataController) {
```
Comment (lbajolet-hashicorp) [inline builder/vsphere/common/step_add_cdrom.go:86]: Very nitpicky, but I would suggest inverting this one to reduce the nesting by one level for the `ErrNoSataController` case :)

**Review by lbajolet-hashicorp — APPROVED** [review]
Thanks for the PR @tenthirtyam,

I've left one tiny nitpick on the type check, but besides that, LGTM!

Pre-approving this one

File: builder/vsphere/common/step_add_cdrom.go (hunk 1 above)
Comment (tenthirtyam) [reply builder/vsphere/common/step_add_cdrom.go]: Updated in 9e0a0d3.

**Review by lbajolet-hashicorp — APPROVED** [review]
Thanks for the reroll @tenthirtyam!

LGTM at first glance, will merge if/when CI goes green :)

**Pull request merged**